tools/debug-ui/src/api.tsx

```diff
@@ -410,29 +410,65 @@ export interface ChainProcessingInfo {
     floating_chunks_info: ChunkProcessingInfo[];
 }
 
+
+// Helper to format the URL through the proxy
+const getProxyUrl = (addr: string, endpoint: string) => {
+    // This requests /api-proxy/address/endpoint relative to current domain
+    return `/api-proxy/${addr}/${endpoint}`;
+};
+
 export interface ChainProcessingStatusResponse {
     status_response: {
         ChainProcessingStatus: ChainProcessingInfo;
     };
 }
 
 export async function fetchBasicStatus(addr: string): Promise<StatusResponse> {
-    const response = await fetch(`http://${addr}/status`);
+    let response: Response;
+    if (window.location.protocol === 'https:') {
+        response = await fetch(getProxyUrl(addr, 'status'));
+    } else if (window.location.protocol === 'http:') {
+        response = await fetch(`http://${addr}/status`);
+    } else {
+        throw new Error("Unsupported protocol: must start with http:// or https://");
+    }
```

Comment:
Can we please make either this part a function like fetchAPI or make a function like `getURL(protocol, addr, endpoint)` to avoid repetition?